tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -47,7 +50,12 @@ FOR_EACH(K_APPMEM_PARTITION_DEFINE, (;), part0, part1);
  */
 struct k_mem_domain dom0;
 struct k_mem_domain dom1;
+struct k_thread user_thread0;
+struct k_queue pqueue;
+
+static struct data_st test_data[LIST_LEN];
 
+K_APP_DMEM(part0) static volatile bool give_uthread_end_sem;
```

Comment:
why was this added? nothing seems to use it in this file unless I'm missing something

Reply:
You are right, maybe during multiple rebasing I forgot to remove it.

Reply:
If you mean 
```
K_APP_DMEM(part0) static volatile bool give_uthread_end_sem;
K_APP_DMEM(part0) bool mem_access_check;
```
yes, it is just legacy and not used anywhere, I removed that.

That structures are used in test `test_access_kernel_obj_with_priv_data`:
```
struct k_thread user_thread0;
struct k_queue pqueue;

static struct data_st test_data[LIST_LEN];
```